File: server/DataCore/invi.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace inv19.models { 
	/* invi -  Запчасть */ 

 public class  invp_data { // Описание
	 public System.Guid  invp_dataId{ get; set; } // Идентификатор (первичный ключ)
	 public List<invp_tag>  invp_tag { get; set; } // дочерний раздел: Метки
	[Required]
	public string  name{ get; set; } // Название
	public System.Guid  groupid { get; set; } //Группа
	
	public System.Guid  subgroupid { get; set; } //Подгруппа

	public System.Guid  departmentid { get; set; } //Отдел
	
	public System.Guid  machineid { get; set; } //Машина
	
 }

 public class  invp_tag { // Метки
	 public System.Guid  invp_tagId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	 public System.Guid  invp_dataId { get; set; } // обратная ссылка на: Описание
	[Required]
	public string  RFID{ get; set; } // Метка RFID
 }
}

File: client/src/app/invi.ts
﻿import { enums } from './enums';

export namespace invi { 
	/* invi -  Запчасть */ 

 export interface   invp_data { // Описание
	invp_dataId:string; // Primary key field
	name:string; // Название
	groupid:string; //Группа -> invg_grp
	subgroupid:string; //Подгруппа -> invg_subgrp
	departmentid:string; //Отдел -> invd_dep
	machineid:string; //Машина -> invd_machine
	// add dereference fields 
	groupid_name :string; // dereference for invg_grp
	subgroupid_name :string; // dereference for invg_subgrp
	departmentid_name :string; // dereference for invd_dep
	machineid_name :string; // dereference for invd_machine
 }

 export interface   invp_tag { // Метки
	invp_tagId:string; // Primary key field
	  invp_dataId:string; // Описание
	rFID:string; // Метка RFID
 }
}